### PR TITLE
Enable IOUPreview for bill splits

### PR DESCRIPTION
**Pull request by Julesssss**
CC @luacmartins @mountiny 

### Details

Enabled the IOUPreview to display within groups. Bill Splits amount, participant, comment, and currency data will show in the preview. Tapping it will open the group participants page for now. 

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/270707

### Tests

**1) Request money from a single user**
- Locate any user chat
- Request money from them via the FAB or ChatReport menu
- An IOUPreview should display
- It should show 'Cash' at the top of the component
- The amount and currency should be displayed
- If a description was added, it should display
- Both user avatars should be displayed
- Hovering over the avatars should display the participant emails
- Tapping it should take you to the IOUDetails Modal

**2) Split a bill with a group of users**
- Locate any group chat with 3 or more users
- Split a bill with the group via the FAB or ChatReport menu
- Multiple 1:1 requests should be created
- An IOUPreview should display within the group chat
- It should show 'Split' at the top of the component
- The amount and currency should be displayed
- If a description was added, it should display
- One avatar should be displayed, with a counter displaying the number of other participants
- Hovering over the avatars should display the participant emails
- Tapping it should take you to the Details page, showing the list of other participants

### Offline tests

Run the above tests. **[FOR TEST 1 ONLY]** In addition, you should see a 'Total will update when you're back online' message in the IOUPReview if your users are on different currencies. This doesn't apply to group splits as we don't have an associated iouReport to read from.
Screenshot 2023-05-03 at 15 44 07

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x]  I linked the correct issue in the `### Fixed Issues` section above
- [x]  I wrote clear testing steps that cover the changes made in this PR
    - [x]  I added steps for local testing in the `Tests` section
    - [x]  I added steps for the expected offline behavior in the `Offline steps` section
    - [x]  I added steps for Staging and/or Production testing in the `QA steps` section
    - [x]  I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x]  I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x]  I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x]  I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x]  I ran the tests on **all platforms** & verified they passed on:
    - [x]  Android / native
    - [x]  Android / Chrome
    - [x]  iOS / native
    - [x]  iOS / Safari
    - [x]  MacOS / Chrome / Safari
    - [x]  MacOS / Desktop
- [x]  I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x]  I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x]  I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x]  I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x]  I verified that comments were added to code that is not self explanatory
    - [x]  I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x]  I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x]  If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x]  I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x]  I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x]  I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x]  I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x]  If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x]  I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x]  I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x]  I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x]  I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x]  I verified that if a function's arguments changed that all usages have also been updated correctly
- [x]  If a new component is created I verified that:
    - [x]  A similar component doesn't exist in the codebase
    - [x]  All props are defined accurately and each prop has a `/** comment above it */`
    - [x]  The file is named correctly
    - [x]  The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x]  The only data being stored in the state is data necessary for rendering and nothing else
    - [x]  For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x]  Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x]  All JSX used for rendering exists in the render method
    - [x]  The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x]  If any new file was added I verified that:
    - [x]  The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x]  If a new CSS style is added I verified that:
    - [x]  A similar style doesn't already exist
    - [x]  The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x]  If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x]  If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x]  If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x]  If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x]  If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x]  I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Screenshot 2023-05-03 at 15 35 11

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Screenshot_20230503_165724

</details>

<details>
<summary>Mobile Web - Safari</summary>

Simulator Screen Shot - iPhone 14 - 2023-05-03 at 16 07 47

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-05-03 at 15 36 09

</details>

<details>
<summary>iOS</summary>

IMG_0133

</details>

<details>
<summary>Android</summary>

Screenshot_20230505-124738

</details>

### CODE REVIEW

File: src/components/MoneyRequestConfirmationList.js
Comment (Julesssss) [inline src/components/MoneyRequestConfirmationList.js:107]: A minor change that was necessary. Else the new 'split' string would need an alternative name.

**Comment by Julesssss** [comment]
Sorry @danieldoglas & @pecanoro, Pullerbear was assigned twice. Please feel free to ignore this review

**Comment by Julesssss** [comment]
Bumping for review, this is a blocker for many Manual Request/EC3 project issues. 

@pecanoro @parasharrajat @mountiny @luacmartins

File: src/components/ReportActionItem/IOUPreview.js
Comment (pecanoro) [inline src/components/ReportActionItem/IOUPreview.js:136]: Just a question since I am not sure how the IOU action is built. When it's a 1:1 action, we don't store the managerEmail and ownerEmail in participants? Just wondering why we can't use participants for both cases.

File: src/components/ReportActionItem/IOUPreview.js
Comment (pecanoro) [inline src/components/ReportActionItem/IOUPreview.js:158]: Can you leave a comment explaining why we get the amount from different sources depending on the type of IOU? For example, I don't exactly what's the difference between `props.iouReport.total` and `props.action.originalMessage.amount`.

File: src/pages/home/report/ReportActionItem.js
Comment (pecanoro) [inline src/pages/home/report/ReportActionItem.js:185]: You can use `?:` for this, right? Not sure if I am not seeing something here.

File: src/components/ReportActionItem/IOUPreview.js
Comment (pecanoro) [reply src/components/ReportActionItem/IOUPreview.js:158]: Ohh I see later that you left a comment saying "// Only 1:1 requests require an iouReport. For splits we rely on reportAction data.", I think you leave a comment saying the same here.

Comment (pecanoro) [reply src/components/ReportActionItem/IOUPreview.js:158]: Either way, instead of this complex if/else in which the code is almost repeated, I think you can simply find the proper total depending on the request type and then maybe call `props.numberFormat` once with the amount.

**Review by pecanoro — DISMISSED** [review]

File: src/components/ReportActionItem/IOUPreview.js
Comment (luacmartins) [inline src/components/ReportActionItem/IOUPreview.js:142]: Can we change this name? cachedTotal is really misleading since we also have a cachedTotal key in the iouReport object (which we should really get rid of)

File: src/components/ReportActionItem/IOUPreview.js
Comment (luacmartins) [reply src/components/ReportActionItem/IOUPreview.js:158]: Agree with only calling numberFormat once

File: src/languages/en.js
Comment (luacmartins) [inline src/languages/en.js:321]: We are changing these to start with lower case letters

File: src/languages/es.js
Comment (luacmartins) [inline src/languages/es.js:320]: Same here. Start with lower case please

**Review by luacmartins — CHANGES_REQUESTED** [review]
Looking good. Left a few comments

File: src/components/ReportActionItem/IOUPreview.js
Comment (Julesssss) [reply src/components/ReportActionItem/IOUPreview.js]: Thanks both, I vastly improved and simplified this logic.

**Review by luacmartins — DISMISSED** [review]
LGTM

File: src/components/ReportActionItem/IOUPreview.js
Comment (Julesssss) [reply src/components/ReportActionItem/IOUPreview.js]: Done

**Comment by Julesssss** [comment]
Thanks, both. I applied changes to all feedback excluding the [iou action logic](https://github.com/Expensify/App/pull/18143/files#r1184792193). Please take a look and share any further concerns!

**Comment by Julesssss** [comment]
I'll add the iOS/Android screenshots tomorrow morning

**Comment by pecanoro** [comment]
Will review it tomorrow morning again so we can merge asap!

**Review by mountiny — COMMENTED** [review]
Code looks good to me but havent tests, @aimane-chnaif and @parasharrajat should be bale to handle that

**Comment by mountiny** [comment]
@aimane-chnaif I believe all these changes do not require any backend changes.

File: src/components/MoneyRequestConfirmationList.js
Comment (Julesssss) [reply src/components/MoneyRequestConfirmationList.js:107]: Ah thanks. We're reusing that in two places so I'll need to duplicate the string.

**Comment by luacmartins** [comment]
@aimane-chnaif we'll have some capitalization issues for now. Also, the `with <users>` part is no longer part of the template so those are expected for now. I think the only real issue is https://github.com/Expensify/App/pull/18143#discussion_r1185346558

**Comment by luacmartins** [comment]
@Julesssss we have some conflicts

File: src/components/MoneyRequestConfirmationList.js
Comment (Julesssss) [reply src/components/MoneyRequestConfirmationList.js]: Done

**Review by luacmartins — DISMISSED** [review]

**Comment by Julesssss** [comment]
Thanks @aimane-chnaif, nice finds. 

I fixed the capitalization issue, fixed conflicts, and fixed another bug where phone number users weren't working properly.

As for the split participant avatars not showing, I tried to resolve this but it's a bit tricky to figure out the correct format for the optimistic data. I created a [separate follow-up issue](https://github.com/Expensify/App/issues/18472) to work on this as it probably shouldn't block the changes.

**Comment by Julesssss** [comment]
ready for your review: @aimane-chnaif @pecanoro @luacmartins 

I think [this thread](https://github.com/Expensify/App/pull/18143#discussion_r1184792193) is the last remaining blocker.

File: src/components/ReportActionItem/IOUPreview.js
Comment (mountiny) [reply src/components/ReportActionItem/IOUPreview.js:140]: I agree with the conclusion from Jules

Comment (pecanoro) [reply src/components/ReportActionItem/IOUPreview.js:140]: Thank you for the explanation!

**Review by pecanoro — DISMISSED** [review]

**Comment by mountiny** [comment]
@Julesssss i am still getting the issue wiht the total amount showing in both of the previews 
image

**Review by mountiny — DISMISSED** [review]
great progress

File: src/components/ReportActionItem/IOUAction.js
Comment (Julesssss) [reply src/components/ReportActionItem/IOUAction.js:101]: Thanks, yeah I think this was also related to a currency conflict from yesterday. I had already fixed this in my next PR but will fix this now 👍

**Comment by Julesssss** [comment]
Okay, sorry about that. Completely removing the `shouldDisplayIOUPreview` logic should have been saved for the next PR. 

Fixed!

File: src/components/ReportActionItem/IOUAction.js
Comment (Julesssss) [inline src/components/ReportActionItem/IOUAction.js:81]: While we could include 'pay' in the array, it doesn't really make sense because it is not a IOURequest. Also, we're going to completely remove this condition with my next PR.

Comment (mountiny) [reply src/components/ReportActionItem/IOUAction.js:81]: We can do that

**Review by pecanoro — APPROVED** [review]

**Review by luacmartins — APPROVED** [review]

**Review by mountiny — CHANGES_REQUESTED** [review]
~requesting $90 in a group of 3 people has split the amount as if it was a group of 4 people~ all good I have added more people to the group than I wanted.... so its 4 actually
image

**Review by mountiny — APPROVED** [review]
All good, I am idiot.

**Comment by luacmartins** [comment]
Just waiting on final review by @aimane-chnaif! Thanks for the urgency on this one!

**Review by aimane-chnaif — APPROVED** [review]

**Comment by Julesssss** [comment]
Thanks for the detailed review @aimane-chnaif. You should claim the bug bounty for those other issues you found too 😉

File: src/components/ReportActionItem/IOUAction.js
Comment (eVoloshchak) [inline src/components/ReportActionItem/IOUAction.js:109]: Coming from https://github.com/Expensify/App/issues/18671.
We were using participants.length to determine if action is a split action, so the check would fail if you delete all of the participants from the workspace.
This was fixed by checking the original type of the action
`const isSplitBillAction = lodashGet(props.action, 'originalMessage.type', '') === CONST.IOU.REPORT_ACTION_TYPE.SPLIT;`

**Pull request merged**
